mayastor-test/test_nexus.js

```diff
@@ -187,6 +216,17 @@ describe('nexus', function() {
         next => {
           fs.truncate(aioFile, diskSize, next);
         },
+        next => {
+          fs.writeFile(uringFile, '', next);
+        },
+        next => {
+          fs.truncate(uringFile, diskSize, next);
+        },
+        next => {
+          if (doUring())
```

Comment:
just a general note, it is possible to skip tests in mocha. In that case, the test is highlighted in the output making it easy for anyone to see that the test exists but wasn't run for some reason. As it is written now, it does not give a clue that some functionality of mayastor was not tested. On the other hand, duplicating current tests just for the purpose of testing io_uring would be probably even worse :-)

Reply:
Replying to myself... :-) Would we prefer to fail the tests instead? I mean if we ever make a mistake and misconfigure our ci/cd pipeline to run on a system that does not support io_uring, I would rather see it to fail (with error message explaining why) than to not notice that something is not tested.

Reply:
I went through this with @gila and there were discussions on whether we could set up a loop device and either use it directly or format it as XFS and use a file there. The problem comes with the clean up stage and, from past experience, failing to clean up properly then leads to intermittent failures later. At the moment, it skips it on CI (where /tmp is overlay fs as it's inside the container) and on Ubuntu (which defaults to ext4), but will run elsewhere like CentOS/RHEL (defaults to XFS) or @gila's machine.